packages/pg-protocol/src/serializer.ts

```diff
@@ -106,50 +106,53 @@ type BindOpts = {
   binary?: boolean
   statement?: string
   values?: any[]
+  // optional map from JS value to postgres value per parameter
+  valueMap?: (param: any) => any
```

Comment:
Rather than "valueMap", how about calling this "mapper" or "transformer"? The "Map" suffix sounds like it's an actual map but it's a function.

Also, this should accept more arguments, including the parameter index being transformed.